rpc/types/src/misc/block_complete_entry.rs

```diff
@@ -25,13 +29,91 @@ pub struct BlockCompleteEntry {
     pub txs: Vec<TxBlobEntry>,
 }
 
-// TODO: custom epee
-// <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/cryptonote_protocol/cryptonote_protocol_defs.h#L138-L163>
+//---------------------------------------------------------------------------------------------------- Serde
 #[cfg(feature = "epee")]
-epee_object! {
-    BlockCompleteEntry,
-    pruned: bool,
-    block: String,
-    block_weight: u64,
-    txs: Vec<TxBlobEntry>,
+/// [`EpeeObjectBuilder`] for [`BlockCompleteEntry`].
+///
+/// Not for public usage.
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct __BlockCompleteEntryEpeeBuilder {
+    pub pruned: Option<bool>,
+    pub block: Option<String>,
+    pub block_weight: Option<u64>,
+    pub txs: Option<Vec<TxBlobEntry>>,
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObjectBuilder<BlockCompleteEntry> for __BlockCompleteEntryEpeeBuilder {
+    fn add_field<B: Buf>(&mut self, name: &str, r: &mut B) -> error::Result<bool> {
+        macro_rules! read_epee_field {
+            ($($field:ident),*) => {
+                match name {
+                    $(
+                        stringify!($field) => { self.$field = Some(read_epee_value(r)?); },
+                    )*
+                    _ => return Ok(false),
+                }
+            };
+        }
+
+        read_epee_field! {
+            pruned,
+            block,
+            block_weight,
+            txs
+        }
+
+        Ok(true)
+    }
+
+    fn finish(self) -> error::Result<BlockCompleteEntry> {
+        const ELSE: error::Error = error::Error::Format("Required field was not found!");
+        Ok(BlockCompleteEntry {
+            pruned: self.pruned.unwrap_or(false),
+            block: self.block.ok_or(ELSE)?,
+            block_weight: self.block_weight.unwrap_or(0),
+            txs: self.txs.ok_or(ELSE)?,
+        })
+    }
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObject for BlockCompleteEntry {
```

Comment:
Sorry for not realizing this earlier but this type is already defined in `cuprate-wire`:

https://github.com/Cuprate/cuprate/blob/0910c0a231b4e0ef48b6cf0b9d7d48a44cc557a7/net/wire/src/p2p/common.rs#L223-L234

I don't really want to move it to `cuprate-types` but that may be the best option?